src/modules/custodian/custodian.service.test.ts

```diff
@@ -93,8 +171,160 @@ describe('CustodianService', () => {
+      when(httpServicePost)
+        .calledWith(...expectedHttpServiceReadJobArgs)
+        .mockReturnValueOnce(
+          httpResponseGenerator([
+            {
+              Started: valueGenerator.dateTimeString(),
+              Completed: valueGenerator.dateTimeString(),
+              Failed: true,
+            },
+          ]),
+        );
+      const response = await service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      expect(res.status).toHaveBeenCalledWith(500);
+      expect(response).toEqual({ folderName, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_FAILED });
+      expect(loggerWarn).not.toHaveBeenCalled();
+    });
+
+    it('returns null if custodian job failed in previous run, and we will allow repeated call', async () => {
+      when(cacheManagerGet).calledWith(cacheId).mockReturnValueOnce({ requestId: custodianRequestId, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_FAILED });
+
+      const response = await service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      expect(res.status).not.toHaveBeenCalled();
+      expect(response).toBeNull();
+      expect(loggerWarn).toHaveBeenCalledWith('Found cache with failed Custodian job for folder creation. Lets try to create new job.');
+    });
+
+    it('throws exception if job is still being sent to Custodian, duplicate call detected', async () => {
+      when(cacheManagerGet).calledWith(cacheId).mockReturnValueOnce({ requestId: custodianRequestId, status: ENUMS.FOLDER_STATUSES.SENDING_TO_CUSTODIAN });
+
+      const responsePromise = service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      await expect(responsePromise).rejects.toBeInstanceOf(BadRequestException);
+      await expect(responsePromise).rejects.toThrow(
+        'Custodian job for folder creation is still being sent to Custodian. Most likely we received multiple calls at same time.',
+      );
+    });
+
+    it('throws exception if reading job from custodian fails', async () => {
+      const axiosError = new AxiosError();
+      const errorBody = { errorMessage: valueGenerator.string() };
+      axiosError.response = {
+        data: errorBody,
+        status: 401,
+        statusText: 'Unauthorized',
+        headers: undefined,
+        config: undefined,
+      };
+      when(cacheManagerGet).calledWith(cacheId).mockReturnValueOnce({ requestId: custodianRequestId, status: ENUMS.FOLDER_STATUSES.SENT_TO_CUSTODIAN });
+      when(httpServicePost)
+        .calledWith(...expectedHttpServiceReadJobArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const responsePromise = service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      await expect(responsePromise).rejects.toBeInstanceOf(CustodianException);
+      await expect(responsePromise).rejects.toThrow(`Failed to get an Provisioning job by request id for cache id ${cacheId}.`);
```

Comment:
`Failed to get an Provisioning job` -> `Failed to get a Provisioning job`

Reply:
Thanks, changed